packages/api-client/src/v2/features/search/components/SearchModal.vue

```diff
@@ -0,0 +1,145 @@
+<script setup lang="ts">
+import {
+  ScalarModal,
+  ScalarSearchInput,
+  ScalarSearchResultList,
+  type ModalState,
+} from '@scalar/components'
+import type { WorkspaceEventBus } from '@scalar/workspace-store/events'
+import type { WorkspaceDocument } from '@scalar/workspace-store/schemas'
+import { nanoid } from 'nanoid'
+import { computed, ref, watch } from 'vue'
+
+import { useSearchIndex } from '@/v2/features/search/hooks/use-search-index'
+
+import SearchResult from './SearchResult.vue'
+
+const props = defineProps<{
```

Comment:
we should always destructure props, good for defaults and I swear better type safety in the template